src/client/tcp.rs

```diff
@@ -26,6 +26,20 @@ pub async fn connect_slave(socket_addr: SocketAddr, slave: Slave) -> Result<Cont
     Ok(context)
 }
 
+/// Connect to a physical, broadcast, or custom Modbus device,
+/// and can enable recover strategy when modbus header mismatch,
+/// when one response frame isn't arrived in specified time due
+/// to tcp retransmission, in the next reading or writing previous
+/// reponse frame will arrived, with this strategy the mismatched
+/// response frame can be discarded. This strategy is inspired by
+/// libmodbus's error recovery: 
+/// https://libmodbus.org/reference/modbus_set_error_recovery/
```

Comment:
When following the link I don't see any connection between the recovery modes offered by libmodus and this parameter. They don't even mention any retires? This is more confusing than it helps.

Reply:
I quickly checked the rodbus implementation and didn't find anything similar.

Please elaborate why this retry is needed and desired. We should not introduce common parameters that only solve a particular use case.

Reply:
In **libmodbus**, discard ready frames when transcation id mismatch and error recover strategy enabled.

```c
        rc = ctx->backend->pre_check_confirmation(ctx, req, rsp, rsp_length);
        if (rc == -1) {
            if (ctx->error_recovery & MODBUS_ERROR_RECOVERY_PROTOCOL) {
                _sleep_response_timeout(ctx);
                modbus_flush(ctx);
            }
            return -1;
        }
``` 

I assume that, first several (1 or 2) frames are mismatch, the ok frame may in the remaining.

For situation in #111 discard one frame, the remaining can work.